scripts/testy_launch.py

```diff
@@ -117,140 +117,147 @@ def snapshot_exists(snapshot_id):
         raise Exit(result.stderr)
     return True if result.stdout.strip() else False
 
-def wait_instance_running(instance_id):
-    print(f"Waiting for the EC2 instance '{instance_id}' to be running ...")
-    # Code 16 corresponds to "running".
-    expected_state = "16"
-    max_retries = 20
-    num_retry = 0
+def wait_on_status_check(instance_id):
+    max_retries = 30
+    sleep_time = 10
+    retry = 0
 
-    while num_retry < max_retries:
-        result = local(f"aws ec2 describe-instance-status \
-            --instance-ids {instance_id} \
-            --query 'InstanceStatuses[*].InstanceState.Code' \
-            --output text", hide=True, warn=True)
-        if result.stderr:
-            raise Exit(result.stderr)
-        state = result.stdout.strip()
-        if state != expected_state:
-            num_retry += 1
-            time.sleep(10)
-        else:
-            break
-    else:
-        raise Exit(f"Timeout: the instance {instance_id} is not running.")
+    timer = cycle(['\\', '|', '/', '—'])
+    msg = f"Waiting for the status check to complete. This may take several minutes ..."
+    print("", end=f"{msg}\r", flush=True)
 
-def wait_instance_status_check_ok(instance_id):
-    print(f"Waiting for the EC2 instance '{instance_id}' to pass the status check ...")
-    max_retries = 20
-    num_retry = 0
+    while retry < max_retries:
 
-    while num_retry < max_retries:
         result = local(f"aws ec2 describe-instance-status \
             --instance-ids {instance_id} \
             --query 'InstanceStatuses[*].[InstanceStatus.Status,SystemStatus.Status]' \
-            --output text", hide=True, warn=True)
+            --output json", hide=True, warn=True)
         if result.stderr:
             raise Exit(result.stderr)
-        checks = result.stdout.strip().split()
-
-        while True:
-            if not len(checks):
-                return
-            if checks[0] != 'ok':
-                break
-            checks.pop(0)
-        num_retry += 1
-        time.sleep(10)
+
+        status = json.loads(result.stdout)
+        if (len(status) == 0 or len(status[0]) != 2):
```

Comment:
It would be great to have a couple of examples describing the potential output to understand the `[x][y]`

Reply:
Updated in [85eced0](https://github.com/wiredtiger/testy/pull/50/commits/85eced09a322317408387c9b43cbe8b4d29c7bf3).